recipe-client-addon/lib/AddonStudies.jsm

```diff
@@ -97,7 +97,7 @@ async function markAsEnded(db, study) {
   study.active = false;
   study.studyEndDate = new Date();
   await getStore(db).put(study);
-  Services.obs.notifyObservers(study, STUDY_ENDED_TOPIC);
+  Services.obs.notifyObservers(study, STUDY_ENDED_TOPIC, `${study.recipeId}`);
```

Comment:
Yeah, the subject is an nsISupports object. You could add a `wrappedJSObject` property that points to itself (`study.wrappedJSObject = study`) and that might work to extract it on the other side, but it's possible you'd need to also define a `QueryInterface` implementation - not sure off-hand.